browser-extension/mv2/src/background/background.js

```diff
@@ -1020,10 +1020,29 @@ BG.Methods.addListenerForExtensionMessages = function () {
 
 BG.Methods.getSessionRecordingConfig = async (url) => {
   const sessionRecordingConfig = await RQ.StorageService.getRecord(RQ.STORAGE_KEYS.SESSION_RECORDING_CONFIG);
+  const isAutoRecordingActive = sessionRecordingConfig?.autoRecording?.isActive;
   const pageSources = sessionRecordingConfig?.pageSources || [];
+
+  if ("autoRecording" in sessionRecordingConfig) {
+    // no pages
+    if (!sessionRecordingConfig?.autoRecording.isActive) {
+      return null;
+    } else if (sessionRecordingConfig?.autoRecording.mode === "allPages") {
+      pageSources.push({
+        value: "*",
+        key: "Url",
+        isActive: true,
+        operator: "Wildcard_Matches",
+      });
```

Comment:
Why to push in the existing `pageSources` array? We can create a new array for the `allPages` config. @lazyvab @rohanmathur91 WDYT?